src/client/chat.js

```diff
@@ -334,53 +334,75 @@ module.exports = function (client, options) {
   let pendingChatRequest
   let lastPreviewRequestId = 0
 
+  function getAcknowledgements () {
+    let acc = 0
+    const acknowledgements = []
+
+    for (let i = 0; i < client._lastSeenMessages.capacity; i++) {
+      const idx = (client._lastSeenMessages.offset + i) % 20
+      const message = client._lastSeenMessages[idx]
+      if (message) {
+        acc |= 1 << i
+        acknowledgements.push(message.signature)
+        message.pending = false
+      }
+    }
+
+    const bitset = Buffer.allocUnsafe(3)
+    bitset[0] = acc & 0xFF
+    bitset[1] = (acc >> 8) & 0xFF
+    bitset[2] = (acc >> 16) & 0xFF
+
+    return {
+      acknowledgements,
+      acknowledged: bitset
+    }
+  }
+
   client._signedChat = (message, options = {}) => {
     options.timestamp = options.timestamp || BigInt(Date.now())
     options.salt = options.salt || 1n
 
-    if (message.startsWith('/') && !mcData.supportFeature('useChatSessions')) {
+    if (message.startsWith('/')) {
       const command = message.slice(1)
-      client.write('chat_command', {
-        command,
-        timestamp: options.timestamp,
-        salt: options.salt,
-        argumentSignatures: signaturesForCommand(command, options.timestamp, options.salt),
-        signedPreview: options.didPreview,
-        previousMessages: client._lastSeenMessages.map((e) => ({
-          messageSender: e.sender,
-          messageSignature: e.signature
-        })),
-        lastRejectedMessage: client._lastRejectedMessage
-      })
+      if (mcData.supportFeature('useChatSessions')) {
+        const { acknowledged } = getAcknowledgements()
+        client.write('chat_command', {
+          command,
+          timestamp: options.timestamp,
+          salt: options.salt,
+          argumentSignatures: signaturesForCommand(command, options.timestamp, options.salt),
```

Comment:
`signaturesForCommand` calls `client.signMessage(signable, ts, salt)`, and `signMessage` accepts `acknowledgements`. Should I pass `acknowledgements` to `signaturesForCommand` to pass to `signMessage`?

Reply:
Where you pass them from is up to you

Reply:
Ok, I fed them through